precompiles/assets-erc20/src/mock.rs

```diff
@@ -46,57 +46,30 @@ pub type Block = frame_system::mocking::MockBlockU32<Runtime>;
 /// be routed to Erc20AssetsPrecompileSet being marked as foreign
 pub const FOREIGN_ASSET_PRECOMPILE_ADDRESS_PREFIX: u32 = 0xffffffff;
 
-/// The local asset precompile address prefix. Addresses that match against this prefix will
-/// be routed to Erc20AssetsPrecompileSet being marked as local
-pub const LOCAL_ASSET_PRECOMPILE_ADDRESS_PREFIX: u32 = 0xfffffffe;
-
 parameter_types! {
 	pub ForeignAssetPrefix: &'static [u8] = &[0xff, 0xff, 0xff, 0xff];
-	pub LocalAssetPrefix: &'static [u8] = &[0xff, 0xff, 0xff, 0xfe];
 }
 
 mock_account!(ForeignAssetId(AssetId), |value: ForeignAssetId| {
 	AddressInPrefixedSet(FOREIGN_ASSET_PRECOMPILE_ADDRESS_PREFIX, value.0).into()
 });
-mock_account!(LocalAssetId(AssetId), |value: LocalAssetId| {
-	AddressInPrefixedSet(LOCAL_ASSET_PRECOMPILE_ADDRESS_PREFIX, value.0).into()
-});
 
 // Implement the trait, where we convert AccountId to AssetID
 impl AccountIdAssetIdConversion<AccountId, AssetId> for Runtime {
 	/// The way to convert an account to assetId is by ensuring that the prefix is 0XFFFFFFFF
 	/// and by taking the lowest 128 bits as the assetId
 	fn account_to_asset_id(account: AccountId) -> Option<(Vec<u8>, AssetId)> {
-		if account.has_prefix_u32(FOREIGN_ASSET_PRECOMPILE_ADDRESS_PREFIX) {
-			return Some((
-				FOREIGN_ASSET_PRECOMPILE_ADDRESS_PREFIX
-					.to_be_bytes()
-					.to_vec(),
-				account.without_prefix(),
-			));
-		}
-
-		if account.has_prefix_u32(LOCAL_ASSET_PRECOMPILE_ADDRESS_PREFIX) {
-			return Some((
-				LOCAL_ASSET_PRECOMPILE_ADDRESS_PREFIX.to_be_bytes().to_vec(),
-				account.without_prefix(),
-			));
-		}
-
-		None
+		Some((
```

Comment:
You broke the logic in a wrong way here, we should still return None if the AccountId is not prefixed by FOREIGN_ASSET_PRECOMPILE_ADDRESS_PREFIX

Reply:
You are right, will fix it :+1: